index.rb

```diff
@@ -19,56 +21,106 @@ def my_select
     result
   end
 
-  def my_all?
-    my_each do |e|
-      return false unless yield(e)
+  def my_all?(*args)
+    if !args[0].nil?
+      my_each { |item| return false unless args[0] == item }
+    elsif block_given?
+      my_each { |item| return false unless yield(item) }
+    elsif args.instance_of?(Class)
+      my_each { |item| return false unless yield(item) != args }
+    elsif args.instance_of?(Regexp)
+      my_each { |item| return false unless args.match(item) }
+    else
+      my_each { |item| return false unless item }
     end
     true
   end
 
-  def my_any?
-    my_each do |e|
-      return true if yield(e)
+  def my_any?(*args)
+    if !args[0].nil?
+      my_each { |item| return true if args[0] == item }
+    elsif block_given?
+      my_each { |item| return true if item }
+    elsif args.instance_of?(Class)
+      my_each { |item| return true if yield(item) == args }
+    elsif args.instance_of?(Regexp)
+      my_each { |item| return true if args.match(item) }
+    else
+      my_each { |item| return true if item == args }
     end
     false
   end
 
-  def my_none?
-    return true unless block_given?
-
-    my_each do |e|
-      return true unless yield(e)
+  def my_none?(arg = nil, &block)
+    my_any?(arg, &block)
+    if block_given?
+      my_each { |item| return false if yield item }
+    elsif args.nil?
+      my_each { |item| return false if item }
+    elsif args.instance_of?(Class)
+      my_each { |item| return false if item.is_a?(args) }
+    elsif args.instance_of?(Regexp)
+      my_each { |item| return false if args.match(item) }
+    else
+      my_each { |item| return false if item == args }
     end
-    false
+    true
   end
```

Comment:
- [ ] Only the 2 cases with blocks passes are passing, kindly check the other test cases and run them locally.